### PR TITLE
Flexible precision for small and negative values

### LINKED ISSUE #413: Too much rounding on histogram when values have little differences
screen shot 2016-07-11 at 17 49 50

this is for values between 0.991 and 1.

### PR DESCRIPTION
**Pull request by saleiva**
close #413

### CODE REVIEW

**Comment by saleiva** [comment]
CR @javisantana 
I cannot make the failing test to pass - fucking js

**Comment by piensaenpixel** [comment]
Lint is broken, i can try tomorrow

**Comment by saleiva** [comment]
retest please

**Comment by saleiva** [comment]
CR @javisantana

File: spec/formatter.spec.js [hunk 1]
```diff
@@ -2,11 +2,16 @@ var formatter = require('../src/formatter');
 
 describe('formatter', function () {
   it('should format numbers', function () {
-    expect(formatter.formatNumber(0)).toBe('0');
+    expect(formatter.formatNumber(0)).toBe(0);
+    expect(formatter.formatNumber(0.71)).toBe('0.710');
```
Comment (javisantana) [inline spec/formatter.spec.js]: why 0.710 and not 0.71 ?

File: src/formatter.js [hunk 2]
```diff
@@ -4,29 +4,36 @@ var d3 = require('d3');
...
 
-  value = format(value) + (unit ? ' ' + unit : '');
+  if (abs_v > 100) {
+    p = 0;
+  } else if (abs_v > 10) {
+    p = 1;
+  } else if (abs_v > 1) {
+    p = 2;
+  } else if (abs_v > 0) {
+    p = 3;
```
Comment (javisantana) [inline src/formatter.js]: if abs_v = 0.0001 then p = 3 so 0.0001.toFixed(3) is 0.000 which does not make any sense.

I'd use a logarigthm here (when under 1) to see how many of them I want, like

Math.ceil(Math.abs(Math.log(0.0001)/Math.log(10))) -> 4



toFixed( 4 + 2)

Comment (saleiva) [inline src/formatter.js]: That would likely not work fine with numbers like 0.3141 where the result would be 1

**Comment by saleiva** [comment]
CR @javisantana

**Comment by javisantana** [comment]
it lacks of a test for small numbers :/

**Comment by saleiva** [comment]
added those @javisantana

**Pull request merged**